src/app/components/UserAssets/Vesting/VestingWithdrawForm.tsx

```diff
@@ -35,12 +36,15 @@ export const VestingWithdrawForm: React.FC<VestingWithdrawFormProps> = ({
   const { value, loading } = useGetUnlockedVesting(
     vesting.staking,
     vesting.vestingContract,
+    vesting.type,
   );
   const [address, setAddress] = useState(account);
 
   const { send, ...tx } = useSendToContractAddressTx(
     vesting.vestingContract,
-    VestingAbi as AbiItem[],
+    (vesting.type === 'fouryear'
```

Comment:
You use this in 6 places, pls extract it to a util function (e.g. `getVestingAbi`) that takes `vestingType` as a parameter and returns the correct ABI.